### PR TITLE
Packaging

### PR DESCRIPTION
**Pull request by hopkiw**
This is based on @iamsubratp PR, but moved common code to functions and tested on all platforms.

### CODE REVIEW

**Comment by hopkiw** [comment]
misnamed secret, misnamed workflow file, yada yada
/packagebuild

**Comment by hopkiw** [comment]
hmm..invalid service account key..?
/packagebuild

**Comment by hopkiw** [comment]
maybe wrong cacerts.
/packagebuild

**Comment by hopkiw** [comment]
getting somewhere. fixed ca-certs, found insufficient permissions. updated role for service account
/packagebuild

**Comment by hopkiw** [comment]
update service account with service account user role...
/packagebuild

**Comment by hopkiw** [comment]
seems to be running this time. v time consuming, though.. :( 
probably shouldn't capture the output from daisy, you don't get any updates in the build log until completion.

**Comment by hopkiw** [comment]
gotten hung in some way, gcs artifacts aren't being updated anymore and without stdout it's difficult to see why.. updated container image with new logic, new perms
/packagebuild

**Comment by hopkiw** [comment]
big progress!
```
[Daisy] Workflow "build-packages" finished
[Daisy] All workflows completed successfully.
```

**Comment by hopkiw** [comment]
actually getting use out of this now and found some issue..

/packagebuild

**Comment by hopkiw** [comment]
there was one case where try_command failed and that somehow didn't trigger the trap on ERR, so the instance was hung forever. updated overall timeout relative to how long i've observed this to take on successful runs, but will want to figure out that case later.

File: packaging/googet/google-osconfig-agent.goospec [hunk 1]
```diff
@@ -30,7 +30,6 @@
       ]
   }],
   "build": {
-    "linux": "/bin/bash",
-    "linuxArgs": ["-c", "GOOS=windows /tmp/go/bin/go build -ldflags='-X main.version={{.version}}' -mod=readonly -o google_osconfig_agent.exe"]
```
Comment (adjackura) [inline packaging/googet/google-osconfig-agent.goospec]: This shouldn't need to be changed as long as you goopack line from packaging/build_goo.sh

File: packaging/build_goo.sh [hunk 2]
```diff
@@ -13,8 +13,18 @@
...
+. packaging/common.sh
+
+echo "Installing go"
+install_go
+
+# Pull go deps
+$GO mod download
+
+# Install dependencies.
+$GO get github.com/google/googet/goopack
```
Comment (adjackura) [inline packaging/build_goo.sh]: Keep the github.com/google/googet/v2/goopack

**Pull request merged**